README.md

```diff
@@ -2,3 +2,31 @@
 a bot to help assist the crates.io, website, and other rust ops teams
 
 [![Build Status](https://travis-ci.com/rust-lang/ops-bot.svg?token=hHeDp9pQmz9kvsgRNVHy&branch=master)](https://travis-ci.com/rust-lang/ops-bot)
+
+**This project is currently under development**
+
+## Intro
+
+This is a chat bot for [Discord](https://discordapp.com/) to manage the infrastructure of [crates.io](https://crates.io/).
+
+Most of the infrastructure for crates.io is hosted by [Heroku](https://www.heroku.com/). Currently if someone
+is to manage any part of the infrastructure for crates.io, they must have full credentials to Heroku. 
+
+The purpose of this bot is to allow basic infrastructure management tasks to be conducted through the Rust Language Discord server.
+It would be limited to specific members of the crates.io team, but it would allow more people to assist with basic infrastructure
+management without needing full credentials to Heroku.
+
+### Stage 1
+
+Stage 1 of this bot will allow an authorized Discord member to:
+* Update select environmental variables (not including things like API keys)
+* Restart the application (which fixes most issues)
+* Scale the application
+* Change Dyno types
+* Deploy the application
+* Rollback a deployment
+* Block/unblock ips
+
+### Future Stages
+
+Once stage 1 is complete, we will consider expanding this bot to be used with other types of infrastructure - i.e. AWS or Microsoft Azure. 
```

Comment:
The functionality listed in stage 1 is not specific to crates.io in any way – it's specific to Heroku. This is a good thing, since it allows to use this bot for other projects as well.

I remember discussions to add crates.io-specific functionality to the bot in the long run, e.g. deleting crate versions. Personally, I'd prefer to put that functionality in an admin interface on crates.io itself, so the bot can remain a more or less generic ops bot, but we can discuss this once we get there.

Reply:
I would actually go the other way-- let's make it crates.io specific first, then maybe extract generic ops bot functionality. The value we need most is to be able to manage crates.io from Discord; I'm not sure how many other people out there want to manage a Heroku app from Discord :) Most of the Rust stuff is on AWS afaik, not much is on Heroku.

Reply:
Specifically, I wouldn't want to give the bot direct access to the database, which I think is problematic in several ways (operations, implementation, security). I think we would be much better off with adding a few admin API endpoints to the crates.io app with well-defined functionality, to maintain proper separation of concerns and privilege isolation.

This isn't really in conflict with adding features that modify the database to the bot as well – the bot could integrate with these API endpoints, or we could implement a separate admin web frontend.